core/src/test/scala/org/apache/spark/scheduler/CoarseGrainedSchedulerBackendSuite.scala

```diff
@@ -184,3 +278,49 @@ class CoarseGrainedSchedulerBackendSuite extends SparkFunSuite with LocalSparkCo
     )
   }
 }
+
+/** Simple cluster manager that wires up our mock backend for the gpu resource tests. */
+private class CSMockExternalClusterManager extends ExternalClusterManager {
+
+  var ts: TaskSchedulerImpl = _
+
+  val MOCK_REGEX = """coarseclustermanager\[(.*)\]""".r
```

Comment:
Just to be tidy, can the members here be private? or use override if they override superclass methods